chain/network/src/peer/peer_actor.rs

```diff
@@ -283,10 +277,8 @@ impl PeerActor {
     }
 
     fn send_message(&self, msg: &PeerMessage) {
-        if let PeerMessage::PeersRequest = msg {
-            if let Some(conn) = &self.connection {
-                conn.last_time_peer_requested.store(self.clock.now());
-            }
+        if let (PeerStatus::Ready(conn), PeerMessage::PeersRequest) = (&self.peer_status, msg) {
+            conn.last_time_peer_requested.store(self.clock.now());
```

Comment:
Here the PeerActor is tracking the last time the PeersRequest has been sent, no matter who tries to send it. I'd prefer to avoid a situation in which NetworkState is managing some field of Connection which PeerActor doesn't touch at all. If anything I'd rather move the "if" statement from ask_for_more_peers to PeerActor, so that PeerActor does the rate limiting on its own. WDYT?

Reply:
Yeah, that's the better way to fix this!

Reply:
done